packages/hierarchy-builder/src/test/values/Formatting.test.ts

```diff
@@ -0,0 +1,68 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { expect } from "chai";
+import { julianToDateTime } from "../../hierarchy-builder/internal/Common";
+import { createDefaultValueFormatter, IPrimitiveValueFormatter } from "../../hierarchy-builder/values/Formatting";
+
+describe("createDefaultValueFormatter", () => {
+  let formatter: IPrimitiveValueFormatter;
+
+  beforeEach(() => {
+    formatter = createDefaultValueFormatter();
+  });
+
+  it("formats id values", async () => {
+    expect(await formatter({ type: "Id", value: "0x123" })).to.eq("0-83");
+  });
+
+  it("formats boolean values", async () => {
+    expect(await formatter({ type: "Boolean", value: false })).to.eq("false");
+    expect(await formatter({ type: "Boolean", value: true })).to.eq("true");
+  });
+
+  it("formats integer values", async () => {
+    expect(await formatter({ type: "Integer", value: 0 })).to.eq("0");
+    expect(await formatter({ type: "Integer", value: 1.23 })).to.eq("1");
+    expect(await formatter({ type: "Integer", value: 7.89 })).to.eq("8");
+  });
+
+  it("formats long values", async () => {
+    expect(await formatter({ type: "Long", value: 0 })).to.eq("0");
+    expect(await formatter({ type: "Long", value: 1.23 })).to.eq("1");
+    expect(await formatter({ type: "Long", value: 7.89 })).to.eq("8");
+    expect(await formatter({ type: "Long", value: Number.MAX_SAFE_INTEGER })).to.eq(Number.MAX_SAFE_INTEGER.toLocaleString());
+  });
+
+  it("formats double values", async () => {
+    expect(await formatter({ type: "Double", value: 0 })).to.eq(Number(0).toLocaleString(undefined, { minimumFractionDigits: 2, maximumFractionDigits: 2 }));
+    expect(await formatter({ type: "Double", value: 1.23 })).to.eq(Number(1.23).toLocaleString());
+    expect(await formatter({ type: "Double", value: 4.56789 })).to.eq(Number(4.57).toLocaleString());
+  });
+
+  it("formats date values", async () => {
+    const date = new Date();
+    expect(await formatter({ type: "DateTime", value: 0 })).to.eq(julianToDateTime(0).toLocaleString());
+    expect(await formatter({ type: "DateTime", value: date.toISOString() })).to.eq(date.toLocaleString());
+    expect(await formatter({ type: "DateTime", value: date })).to.eq(date.toLocaleString());
+    expect(await formatter({ type: "DateTime", extendedType: "ShortDate", value: date })).to.eq(date.toLocaleDateString());
+  });
+
+  it("formats point2d values", async () => {
+    expect(await formatter({ type: "Point2d", value: { x: 1, y: 2 } })).to.eq(
+      `(${await formatter({ type: "Double", value: 1 })}, ${await formatter({ type: "Double", value: 2 })})`,
```

Comment:
Why not compare to specific string `...to.eq("(1.00, 2.00)")`?

Reply:
the decimal separator may be `,` in some locales, so hardcoding it as `.` may cause tests to fail on some machines